src/commands/convertConnections/SelectConnectionsStep.ts

```diff
@@ -0,0 +1,114 @@
+        } else {
+            picks = await this.getRemoteQuickPics(context);
+        }
+
+        if (picks.length === 0) {
+            const noItemFoundMessage: string = localize('noConnectionsFound', 'No connections found in local settings');
+            (await context.ui.showQuickPick(picks, {
+                placeHolder: localize('selectConnections', 'Select the connections you want to convert'),
+                suppressPersistence: true,
+                noPicksMessage: noItemFoundMessage
+            }));
+        } else {
+            context.connections = (await context.ui.showQuickPick(picks, {
+                placeHolder: localize('selectConnections', 'Select the connections you want to convert'),
+                suppressPersistence: true,
+                canPickMany: true,
+            })).map(item => item.data);
+        }
+    }
+
+    public shouldPrompt(context: IConvertConnectionsContext): boolean {
+        return !context.connections || context.connections.length === 0;
+    }
+
+    private async getLocalQuickPics(context: IConvertConnectionsContext, workspaceFolder?: vscode.WorkspaceFolder): Promise<IAzureQuickPickItem<Connection>[]> {
+        const picks: IAzureQuickPickItem<Connection>[] = [];
+        const message: string = localize('selectLocalSettings', 'Select the local settings to convert.');
+        const localSettingsPath: string = await getLocalSettingsFile(context, message, workspaceFolder);
+        const localSettingsUri: vscode.Uri = vscode.Uri.file(localSettingsPath);
+        context.localSettingsPath = localSettingsPath;
+
+        if (await AzExtFsExtra.pathExists(localSettingsPath)) {
+            let localSettings: ILocalSettingsJson = <ILocalSettingsJson>await AzExtFsExtra.readJSON(localSettingsPath);
+            if (localSettings.IsEncrypted) {
+                await decryptLocalSettings(context, localSettingsUri);
+                try {
+                    localSettings = await AzExtFsExtra.readJSON<ILocalSettingsJson>(localSettingsPath);
+                } finally {
+                    await encryptLocalSettings(context, localSettingsUri);
+                }
+            }
+
+            if (localSettings.Values) {
+                for (const [key, value] of Object.entries(localSettings.Values)) {
+                    if (!convertibleSetting(key, value)) {
+                        continue;
+                    }
+
+                    picks.push({
+                        label: key,
+                        data: {
+                            name: key,
+                            value: value
+                        }
+                    });
+                }
+            }
+        }
+
+        return picks;
+    }
+
+    private async getRemoteQuickPics(context: IConvertConnectionsContext): Promise<IAzureQuickPickItem<Connection>[]> {
+        const picks: IAzureQuickPickItem<Connection>[] = [];
+
+        if (context.functionapp) {
```

Comment:
Similar thing to what I had said in another step, but I'd prefer you use nonNullValue here because otherwise, it'll just return an empty array and we won't know if that's because there is no function app or if it's because the function app isn't listing app settings correctly.